tests/basic/storage_pool_test.go

```diff
@@ -10946,3 +10946,133 @@ var _ = Describe("{PoolResizeInTrashCanNode}", func() {
+				log.FailOnError(err, "Failed to run pxctl status on node: %v", node.Name)
+				log.Infof("pxctl status output: %v", output)
+
+				if !strings.Contains(output, "Journal Device") {
+					selectedNode = node
+					break
+				}
+			}
+			if selectedNode.Name == "" {
+				log.FailOnError(fmt.Errorf("No node found without journal device"), "No node found without journal device")
+			}
+			log.Infof("Selected Node for add journal device: %v", selectedNode.Name)
+		})
+		stepLog = "Check lsblk state before addition of drive add of journal device"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			//run lsblk command to check the state of the journal device
+			cmd := "lsblk"
+			lsblkOutput, err = runCmd(cmd, selectedNode)
+			log.FailOnError(err, "Failed to run command: %v", cmd)
+			log.InfoD("lsblk output: %v", lsblkOutput)
+
+		})
+
+		stepLog = "Add a cloud drive as a journal device without pool in maintenance mode"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			//Add a cloud drive as a journal device without pool in maintenance mode
+			driveSpecs, err := GetCloudDriveDeviceSpecs()
+			log.FailOnError(err, "Error getting cloud drive specs")
+
+			deviceSpec := driveSpecs[0]
+			devicespecjournal := deviceSpec + " --journal"
+
+			err = Inst().V.AddCloudDrive(&selectedNode, devicespecjournal, -1)
+			if err == nil {
+				log.FailOnError(fmt.Errorf("adding cloud drive with journal expected ? Error: [%v]", err),
+					"adding cloud drive with journal failed ?")
+			}
+			log.InfoD("adding journal failed as expected. verifying the error")
+
+			re := regexp.MustCompile(".*Requires pool maintenance mode*")
+			if !re.MatchString(err.Error()) {
+				log.FailOnError(err, "Requires pool maintenance mode alert is not received")
+			}
+			//Check lsblk output this should be same as the old output
+			time.Sleep(30 * time.Second)
+			cmd := "lsblk"
+			output, err := runCmd(cmd, selectedNode)
+			log.FailOnError(err, "Failed to run command: %v", cmd)
+			log.InfoD("lsblk output after trying drive add in pool maintenance mode: %v", output)
+			if output != lsblkOutput {
+				log.FailOnError(fmt.Errorf("lsblk output after trying drive add in pool maintenance mode is not same as the old output"), "lsblk output not consistent with old output")
+			}
+		})
+
+		stepLog = "Put pool in maintenance mode and add a cloud drive as a journal device"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			//Put pool in maintenance mode
+			defer func() {
+				err := ExitPoolMaintenance(selectedNode)
+				log.FailOnError(err, "Failed to exit pool maintenance mode")
+			}()
+			err := EnterPoolMaintenance(selectedNode)
+			log.FailOnError(err, "Failed to enter pool maintenance mode")
+
+			//Add a cloud drive as a journal device
+			driveSpecs, err := GetCloudDriveDeviceSpecs()
+			log.FailOnError(err, "Error getting cloud drive specs")
+
+			deviceSpec := driveSpecs[0]
+			devicespecjournal := deviceSpec + " --journal"
+
```

Comment:
How about we run lsblk again before adding cloud drive?

Reply:
Why do we need to run it just before adding? because we are running lsblk initially as well so this would not be needed?

Reply:
My bad. Makes sense. Since we fail the test case if the lsblk output changes.